scripts/contributions/watch_replay.py

```diff
@@ -0,0 +1,1136 @@
+		else:
+			self.data = data
+			line1 = nulls + self.data[0][x_0:frame]
+			line2 = nulls + self.data[1][x_0:frame]
+
+		self.ax.set_xticklabels(list(range(x_0, frames+1, 20)))		# updates the x_labels
+
+		self.lines[0].set_ydata(line1)								# set the data for line1
+		self.lines[1].set_ydata(line2)								# set the data for line2
+
+
+# this class contains all information regarding the entire window
+class Graph:
+	def __init__(self, data, frames_in_turn, healths, save='', fh=None):
+
+		# pretty clear, if no data, raise an Error
+		if len(data) < 1:
+			raise RuntimeError('no data')
+
+		self.fh = fh 																# reference to file handler
+		self.real_time = False if self.fh == None else True 						# tracks whether real-time
+
+		plt.style.use('dark_background')											# sets black background
+
+		plt.rcParams["figure.figsize"] = (16,8)										# resizes the window
+		self.fig, ax = plt.subplots(nrows=1, ncols=2)								# splits the plot into two halves and gets the references
+		self.board_ax, self.info_ax = ax 											# assign left and right side references
+		self.plot_ax = self.fig.add_subplot(324)									# add the plot and assing it's reference
+
+		self.general_init(data, frames_in_turn, healths)							# handles general initialization (called again if in real-time)
+
+		self.head = (0,-1)															# tracks the current turn, frame pair
+		self.end_of_game = False													# end of game flag
+		self.is_manual = False														# stores whether the user is manually moving the slider, keyboard, etc
+		self.single_advance = False													# true when user is scrubbing, but still want to move forward one frame
+		self.stop_slider_evt = False												# stop the slider event from triggereing when the code changes it
+
+		self.patches = PatchWrapper()												# creates the PatchWrapper object
+
+		self.stream = self.data_stream()											# gets a data_reference - this passes all data to the animation
+
+		self.setup_board()															# initialize static parts of the board
+
+		self.fig.canvas.mpl_connect('key_press_event', self.keyboard_input)			# connect keyboard events to the keyboard_input function
+
+		# if in real-time, use a generator function to update number of frames, otherwise frames is static
+		if not self.real_time:
+			self.anim = animation.FuncAnimation(self.fig, self.update, init_func=self.init, frames=self.num_frames, interval=100, blit=BLIT, repeat=False)
+		else:
+			self.frame_generator = self.gen_frames()
+			self.anim = animation.FuncAnimation(self.fig, self.update, init_func=self.init, frames=self.frame_generator, interval=100, blit=BLIT, repeat=False)
+
+		self.change_play_speed('3')													# initialize the playback speed ('3' is default)
+
+		# if you don't save, show the plot. Otherwise save the animation (nothing is shown)
+		if save == '':
+			self.show()
+		else:
+			print ('This may take a little while\nSaving file...')
+			self.anim.save(save)
```

Comment:
```suggestion
			try:
				self.anim.save(save)
			except ValueError:
				sys.stderr.write('FFMPEG is not installed on your system or not in PATH.')
```

Reply:
You should actually just be able to "click to accept". In the other pull request, you changed all the variables again yourself.

Reply:
Yeah, I realized that after I had already gone through and changed everything. But this one I'm actually going to do slightly differently.

This is exactly the sort of thing I was hoping someone would find, I didn't even think about ffmpeg since I already had it installed.

Thanks, I'll also update the header instructions.